### PR TITLE
[macOS] Bring up "flutter_gallery" devicelab, start up test for x86.

### LINKED ISSUE #110084: [macOS] Bring up "flutter_gallery" devicelab, startup tests for x86.

### PR DESCRIPTION
**Pull request by a-wallen**
#### Related
fixes https://github.com/flutter/flutter/issues/110084

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by yaakovschectman — APPROVED** [review]

**Comment by jmagman** [comment]
This broke the tree: https://github.com/flutter/flutter/pull/110372.
Please wait for approval from codeowners before merging.

File: dev/devicelab/bin/tasks/flutter_gallery_macos__start_up.dart
Comment (jmagman) [inline dev/devicelab/bin/tasks/flutter_gallery_macos__start_up.dart:10]: `dev/devicelab/bin/tasks/flutter_gallery_mac__start_up.dart` test was for Android `Mac_android flutter_gallery_mac__start_up`.  The rename to `flutter_gallery_macos__start_up.dart` broke that test.

https://github.com/flutter/flutter/blob/4770e348595551832a4762ea0eabe2cca59b668c/.ci.yaml#L3084

**Comment by christopherfujino** [comment]
Looks like we didn't need this, since `Mac_android flutter_gallery_mac__start_up` already covers it

**Pull request merged**
